### ISSUE TITLE
Allow vue-create Presets to Target Github Enterprise

### ISSUE THREAD

**Issue by matsieftw** [vuejs/vue-cli#3384]
### What problem does this feature solve?
My company uses github enterprise so the preset we want to target is not located on github.com, gitlab.com, or bitbucket.com.

I would like to be able to run the command `vue create --preset <url to my company's private github instance>` or some other mechanism to indicate the domain.

### What does the proposed API look like?
`vue create -preset <domain>/repo_name`

**Comment by matsieftw** [comment]
We are aware of and are using the workaround of having the generator repo pulled down locally, but this means that the user will need to manage their local repo in order to use the generator. By adding a way to set up the domain to use Github Enterprise, we can ensure developers are consistently using the same, most up-to-date version of the generator.

**Comment by natiz** [comment]
+1

**Comment by sodatea** [comment]
I think it's already supported. Since this feature is built on top of `download-git-repo`, it's possible to use urls in the form of `direct:ssh://git@my-gitlab-server.com/group/projectname.git`. There's also an example on the documentation page now:

```
# self-hosted repos
vue create --preset gitlab:my-gitlab-server.com:group/projectname --clone my-project
vue create --preset direct:ssh://git@my-gitlab-server.com/group/projectname.git --clone my-project
```

**Comment by matsieftw** [comment]
So far, that command has not worked for me. I'm not using Gitlab. I am using Github enterprise. Using the direct ssh method does not appear to work.

I get `Failed fetching remote preset` and `Error: 'git clone' failed with status 128`

**Comment by sodatea** [comment]
Is the `--clone` flag correctly added?
If so, it may be a bug in https://github.com/flipxfx/download-git-repo

**Comment by matsieftw** [comment]
Yes. The clone flag is correctly added. This feature does not work.

**Comment by matsieftw** [comment]
Can this ticket be re-opened since it wasn't actually fixed and was closed before any confirmation was made over whether it was fixed?

**Comment by sodatea** [comment]
You should rather report it to download-git-repo, though.

**Comment by sodatea** [comment]
And I personally don't have access to a GitHub enterprise server, so there's much I can help you with.

**Comment by eliranmal** [comment]
for whom it may concern: as a workaround, i created a [CLI tool][1] to ease the process of cloning enterprise/private hosted templates/presets and storing them locally.

@matsieftw @sodatea


[1]: https://github.com/eliranmal/vue-cli-template-registry
